monitoring/collectors/jenkins_collector_adapter.py

```diff
@@ -0,0 +1,59 @@
+import requests
+from collector import ICollector
+from stats import BuildStats, StageStats
+
+
+class InvalidResponseCodeError(Exception):
+    pass
+
+
+class JenkinsCollectorAdapter(ICollector):
+
+    def __init__(self, job_name, build_url):
+        self.job_name = job_name
+        self.url = build_url
+
+
+    def collect(self):
+        resp = self.get_raw_stats_from_jenkins()
+        return self.convert_raw_stats_to_build_stats(resp)
+
+
+    def get_raw_stats_from_jenkins(self):
+        endpoint = self.get_build_stats_endpoint(self.url)
+        resp = requests.get(endpoint)
+
+        if resp.status_code != 200:
+            raise InvalidResponseCodeError()
+
+        return resp.json()
+
+
+    def convert_raw_stats_to_build_stats(self, raw_stats):
+        timestamp = int(raw_stats['endTimeMillis'] / 1000)
+
+        stages_stats = []
+
+        if 'stages' in raw_stats.keys():
+            stages_stats = [StageStats(
+                name = raw_stage['name'],
+                status = raw_stage['status'],
+                duration_millis = raw_stage['durationMillis']
+            ) for raw_stage in raw_stats['stages']]
+
+        build_stats = BuildStats(
+            job_name = self.job_name,
+            build_url = self.url,
+            build_id = raw_stats['id'],
+            finished_at_secs = timestamp,
+            status = raw_stats['status'],
+            duration_millis = raw_stats['durationMillis'],
+            stages = stages_stats,
+        )
+
+        return build_stats
+
+
+    @classmethod
+    def get_build_stats_endpoint(cls, build_url):
```

Comment:
get_build_stats_endpoint should probably be private and not tested directly as well. 

If we want to test it, let's write tests that instantiate JenkinsCollectorAdapter with different URLs and see if collect() triggers a request received by mocked API endpoint at the specified URLs, e.g.:

```
j = JenkinsCollectorAdapter(build_url="wp.pl/myJob/.../x", job_name="myJob")
with request_mock_assert_receives("wp.pl/myJob/.../x"):
   j.collect()
```

Reply:
also, then we don't see a reason for this being a `@classmethod `